### ISSUE TITLE
v1.13.3 - PX4 Gazebo build broken - Ubuntu 22

### ISSUE THREAD

**Issue by satzguyz** [PX4/PX4-Autopilot#21628]
Issue Description:
Unable to start gz_x500 or gazebo simulation targets, in latest version (1.13.3) of px4 using Ubuntu 22 OS as development environment.  But jmavsim works as usual. 

Terminal Command:
make px4_sitl gz_x500

Logs:
ninja: error: unknown target 'gz_x500'
make: *** [Makefile:232: px4_sitl] Error 1

**Detailed Description:**
OS Version : Ubuntu 22
Machine Capacity: intel i9 64 bit machine

Steps to reproduce:
1. clone the master codebase
 git clone https://github.com/PX4/PX4-Autopilot.git --recursive

2. Checkout the v1.13.3 and update submodules
git switch -c v1.13.3
make submodulesclean

3. Install dependent packages
  bash ./Tools/setup/ubuntu.sh

4. Compile for gazebo gz_x500 Target
make px4_sitl gz_x500

Logs as follows,
-- PX4 version: v1.13.3
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.10.6", minimum required is "3") 
-- PX4 config file: /home/my/test/PX4-Autopilot/boards/px4/sitl/default.px4board
-- PLATFORM posix
-- ROMFSROOT px4fmu_common
-- TESTING y
-- ETHERNET y
-- PX4 config: px4_sitl_default
-- PX4 platform: posix
-- PX4 lockstep: enabled
-- cmake build type: RelWithDebInfo
-- The CXX compiler identification is GNU 11.3.0
-- The C compiler identification is GNU 11.3.0
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Building for code coverage
-- ROMFS: ROMFS/px4fmu_common
-- Configuring done
-- Generating done
-- Build files have been written to: /home/my/test/PX4-Autopilot/build/px4_sitl_default
ninja: error: unknown target 'gz_x500'
make: *** [Makefile:235: px4_sitl] Error 1

**Comment by Jaeyoung-Lim** [comment]
@satzguyz Please provide more information on your system. e.g. what your operating system version is, the full log of your terminal and what simulator is installed and what is not installed.

For example, Ubuntu 22.04 primarily supports Gazebo (`gz`) but not Gazebo classic(`gazebo-classic`). With the limited information you have provided, it is hard to reproduce the problem you are running into.

**Comment by satzguyz** [comment]
I am new to this. Kindly point me the template reference which I can follow in Future.

**Comment by Jaeyoung-Lim** [comment]
@satzguyz I see the issue there. v1.13.3 does not support the new gazebo, which is what you are trying to run with `gz`.

If you want to use the new gazebo, you should use the master branch of PX4 (don't skip to v1.13.3)

**Comment by junwoo091400** [comment]
@satzguyz In case you are interested, here's how I got the v1.13.2 to work with Ignition Gazebo 6:

1. Update ignition simulation submodule to include this change: https://github.com/Auterion/px4-simulation-ignition/pull/26
2. Remove the force to use gazebo 5 for ignition from this change: https://github.com/PX4/PX4-Autopilot/pull/19225

Then, `make px4_sitl ignition` should work, if you have the ignition installed.

I did this to test if the current ignition (somehow different from old ignition "edifice" that is now retired) works with v1.13.2. And seems like changing that dependency in submodule & changing the script does the job of using the gazebo 6 ignition (whatever it means).

![image](https://github.com/PX4/PX4-Autopilot/assets/23277211/f06c28b3-fe53-4fde-99d9-14c39e201760)
